### PR TITLE
tools: update our release scripts

### LINKED ISSUE #6472: runtime feature guarding deprecation
(tracking bug to finish up the bits not done in #5693)

### PR DESCRIPTION
**Pull request by alyssawilk**
Functionally this 
- removes the script which filed bugs to remove old config (per plan of record in #6271)
- adds a script which files bugs to remove old code (cleanup of runtime guarded features)
- updates our release process

In practice, this just hot swaps what our deprecation script does and updates the docs.

Risk Level: n/a (tooling + comment)
Testing: manual testing.  Will do one more manual test with actual bug filing just before submitting.
Docs Changes: updated
Release Notes: n/a
Fixes #6472

### CODE REVIEW

File: tools/deprecate_version/deprecate_version.py [hunk 1]
```diff
@@ -72,77 +65,116 @@ def GetConfirmation():
...
+  """Returns a list of tuples of [runtime features to deprecate, PR the feature was added]
+  """
+  repo = Repo(os.getcwd())
+
+  runtime_already_true = GetRuntimeAlreadyTrue()
+
+  # grep source code looking for reloadable features which are true to find the
+  # PR they were added.
+  grep_output = subprocess.check_output(
+      'grep -r "envoy.reloadable_features\." source/*', shell=True)
```
Comment (lizan) [inline tools/deprecate_version/deprecate_version.py]: no need `/*` at end of grep

File: tools/deprecate_version/deprecate_version.py [hunk 2]
```diff
@@ -20,24 +19,18 @@
...
 from collections import defaultdict
 import os
 import re
+import subprocess
 import sys
 
 import github
 from git import Repo
 
-try:
```
Comment (lizan) [inline tools/deprecate_version/deprecate_version.py]: did we move completely to python3 on this?

Comment (alyssawilk) [reply tools/deprecate_version/deprecate_version.py]: I thought we'd switched our tools to 3 by default, and I handled a bunch of string-bytestring compat issues I think were Python 3, and it seems to work without this try phrase.  I can add it back if that's not enough?

Comment (lizan) [reply tools/deprecate_version/deprecate_version.py]: From https://github.com/envoyproxy/envoy/blob/master/tools/deprecate_version/deprecate_version.sh I don't think it is forced to run under Python 3? Python 2 `input` is interpreting the input as python script, (i.e. `eval(raw_input())`), so if not sure let's keep this. Or force Python 3 in the shell script as this one is likely only run by maintainer.

**Pull request merged**